### PR TITLE
Negative volume bug

### PR DESCRIPTION
**Pull request by shellshocked2003**
Fixes bug where negative volumes were being calculated. Error was related to vertex ordering - vertices would be in clockwise instead of counter-clockwise orientation when `normalVec` of a `Face` contained a negative value for the z-component. Only special treatment is required when `normalVec` is the -z axis, in all other cases `rotationMatrix` takes care of rotating vertices to correct orientation. 
Also, loosened constraints in `SeedJointSelector` - too tight `THRESHOLD` was leading to bad load balancing in cases where real joint data was processed.

### CODE REVIEW

File: src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala [hunk 1]
```diff
@@ -160,12 +165,14 @@ object SeedJointSelector {
       } else {
         // Joint meets criteria
         val remainingVolume = jointOption.get
-        if (remainingVolume.volume <= volumePerPiece) {
+        val nonRedundantVolume = Block(remainingVolume.center, remainingVolume.nonRedundantFaces)
```
Comment (jhkolb) [inline src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala:168]: Can you explain why you need to eliminate the non-redundant faces here? Does it really matter when we do this during the normal rock slicing process anyways?

Comment (shellshocked2003) [reply src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala:168]: Unfortunately, we have to eliminate non-redundant faces every time we want to find the volume of a block - simplex integration needs oriented vertices and to find the vertices all redundant faces need to be removed.

Comment (jhkolb) [reply src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala:168]: Okay, let me ask another question to see if I'm understanding this change correctly.
Before, we used the volume of the remaining rock mass to determine what to do, but now we also need the volume of the new seed block we've just created.
That's why we didn't have this code to remove non-redundant faces before, but now we need it, right?

Comment (shellshocked2003) [reply src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala:168]: Yeah, that's essentially it. Additionally, we're using more than one joint set to generate the seed blocks. Depending on how many joint sets are used, there can end up being quite a few redundant faces. The redundant faces lead to vertices being added that aren't part of the block - this throws off the volume calcs.

**Pull request merged**